src/main/groovy/org/shipkit/internal/gradle/versionupgrade/UpgradeDownstreamPlugin.java

```diff
@@ -63,13 +64,22 @@ public void apply(final Project project) {
 
         upgradeDownstreamExtension = project.getExtensions().create("upgradeDownstream", UpgradeDownstreamExtension.class);
 
-        final Task performAllUpdates = TaskMaker.task(project, "upgradeDownstream", new Action<Task>() {
+        final Task performAllUpdates = TaskMaker.task(project, UPGRADE_DOWNSTREAM_TASK, new Action<Task>() {
             @Override
             public void execute(final Task task) {
                 task.setDescription("Performs dependency upgrade in all downstream repositories.");
             }
         });
 
+        ShipkitExecTask ciPerformReleaseTask = (ShipkitExecTask) project.getTasks().findByName(CiReleasePlugin.CI_PERFORM_RELEASE_TASK);
+
+        if(ciPerformReleaseTask != null){
+            ciPerformReleaseTask.getExecCommands().add(execCommand(
+                "Upgrading downstream projects", asList("./gradlew", UPGRADE_DOWNSTREAM_TASK)));
+        } else {
+            throw new IllegalStateException("Please apply org.shipkit.ci-release or org.shipkit.java plugin before applying upgrade-downstream.");
```

Comment:
I wanted to throw this exception, if there is no task, to tell the user what exactly the problem is. But with the split to "ci-upgrade-downstream" and "upgrade-downstream" that you suggested, it is not needed any more.